src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -606,26 +606,27 @@ public ResponseEntity<List<StudentParticipation>> getAllParticipationsForCourse(
             exercise.setGradingInstructions(null);
             exercise.setDifficulty(null);
             exercise.setMode(null);
-            if (exercise instanceof ProgrammingExercise programmingExercise) {
-                programmingExercise.setSolutionParticipation(null);
-                programmingExercise.setTemplateParticipation(null);
-                programmingExercise.setTestRepositoryUri(null);
-                programmingExercise.setShortName(null);
-                programmingExercise.setPublishBuildPlanUrl(null);
-                programmingExercise.setProgrammingLanguage(null);
-                programmingExercise.setPackageName(null);
-                programmingExercise.setAllowOnlineEditor(null);
-            }
-            else if (exercise instanceof QuizExercise quizExercise) {
-                quizExercise.setQuizQuestions(null);
-                quizExercise.setQuizPointStatistic(null);
-            }
-            else if (exercise instanceof TextExercise textExercise) {
-                textExercise.setExampleSolution(null);
-            }
-            else if (exercise instanceof ModelingExercise modelingExercise) {
-                modelingExercise.setExampleSolutionModel(null);
-                modelingExercise.setExampleSolutionExplanation(null);
+            switch (exercise) {
```

Comment:
Didn't know this was possible. Very nice. Gotta remember this one!

Reply:
I think this is new in JDK 21